vllm/entrypoints/openai/rpc/server.py

```diff
@@ -30,9 +30,7 @@ def __init__(self, async_engine_args: AsyncEngineArgs,
 
         # Init socket for readiness state.
         self.socket = self.context.socket(zmq.constants.ROUTER)
-        # Note numeric form of localhost should be used for zmq bind(),
-        # see https://stackoverflow.com/a/8958414
-        self.socket.bind(f"tcp://127.0.0.1:{port}")
+        self.socket.bind(rpc_path)
```

Comment:
How to get a random path:
```python
import zmq
context = zmq.Context()
socket = context.socket(zmq.REP)
socket.bind("ipc://*")
print(socket.getsockopt(zmq.LAST_ENDPOINT)) # ipc:///var/folders/vm/ssf622nn02j77t14q1j8_88w0000gn/T/tmp3sOx6s/socket
```

Reply:
But `getsockopt` will run in the Server process

Reply:
you can switch the `bind/connect`. I believe you can call `.bind` in the client process too.

Reply:
thats a good idea

Reply:
I think this is okay because I used DEALER-DEALER

Reply:
Oh wait, we cannot do this

Reply:
I guess I could just have an additional long lived open socket

Reply:
That's more complicated than I thought. Then it's fine to keep the current one.

Reply:
Cool